mirage/views/osf-resource.ts

```diff
@@ -147,3 +147,61 @@ export function osfNestedResource<K extends keyof ModelRegistry>(
         }
     }
 }
+
+export function osfM2MRelationshipResource<K extends keyof ModelRegistry>(
+    server: Server,
+    parentModelName: K,
+    relationshipName: string & RelationshipsFor<ModelRegistry[K]>,
+    options?: Partial<NestedResourceOptions>,
+) {
+    const opts: NestedResourceOptions = Object.assign({
+        path: `/${pluralize(underscore(parentModelName))}/:parentID/relationships/${underscore(relationshipName)}`,
+        relatedModelName: relationshipName,
+        defaultSortKey: '-id',
+        views: {},
+    }, options);
+    const mirageParentModelName = pluralize(camelize(parentModelName));
+    const actions = gatherActions(opts);
+
+    if (actions.includes('index')) {
+        const relationshipRelatedPath = opts.path.replace('relationships/', '');
+        server.get(relationshipRelatedPath, function(schema, request) {
+            const data = schema[mirageParentModelName]
+                .find(request.params.parentID)[relationshipName]
+                .models
+                .filter((m: ModelInstance) => filter(m, request))
+                .map((model: ModelInstance) => this.serialize(model).data);
+            return process(schema, request, this, data, { defaultSortKey: opts.defaultSortKey });
+        });
+    }
+
+    if (actions.includes('create')) {
```

Comment:
Stop me if you think this is excessive, but these action names don't make as much sense for relationships. I'd expect something like `'self' | 'related' | 'add' | 'remove'`, since you're not creating or deleting resources, and there are two sort-of "index" views... but that'll have cascading changes and force defining more of our own types instead of using mirage's `resourceAction`, and a separate `RelationshipOptions` instead of reusing `ResourceOptions`... 

I think it would make the helper nicer to use, but we could also circle back and do all that later. Your call!